cg/meta/workflow/nf_analysis.py

```diff
@@ -312,22 +313,43 @@ def get_deliverables_template_content() -> list[dict]:
         """Return deliverables file template content."""
         raise NotImplementedError
 
-    def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
-        """Return PipelineDeliverables for a given case."""
-        deliverable_template: list[dict] = self.get_deliverables_template_content()
-        sample_id: str = self.status_db.get_samples_by_case_id(case_id).pop().internal_id
+    def get_deliverables_for_sample(
+        self, sample: Sample, case_id: str, template: list[dict]
+    ) -> list[FileDeliverable]:
+        """Return a list of FileDeliverables for each sample."""
+        sample_id: str = sample.internal_id
+        sample_name: str = sample.name
+        case_path = str(self.get_case_path(case_id=case_id))
         files: list[FileDeliverable] = []
-        for file in deliverable_template:
+
+        for file in template:
+            deliverables = file.copy()
             for deliverable_field, deliverable_value in file.items():
                 if deliverable_value is None:
                     continue
-                file[deliverable_field] = file[deliverable_field].replace("CASEID", case_id)
-                file[deliverable_field] = file[deliverable_field].replace("SAMPLEID", sample_id)
-                file[deliverable_field] = file[deliverable_field].replace(
-                    "PATHTOCASE", str(self.get_case_path(case_id=case_id))
+                deliverables[deliverable_field] = (
+                    deliverables[deliverable_field]
+                    .replace("CASEID", case_id)
+                    .replace("SAMPLEID", sample_id)
+                    .replace("SAMPLENAME", sample_name)
+                    .replace("PATHTOCASE", case_path)
```

Comment:
This could be a fucntion by itself like this one (feel free to modify the function name to what you think works best):
```python
def get_formatted_file_deliverable(
    file_template: dict[str | None, str | None],
    case_id: str,
    sample_id: str,
    sample_name: str,
    case_path: str,
) -> FileDeliverable:
    """Return the formatted file deliverable with the case and sample attributes."""
    deliverables = file_template.copy()
    for deliverable_field, deliverable_value in file_template.items():
        if deliverable_value is None:
            continue
        deliverables[deliverable_field] = (
            deliverables[deliverable_field]
            .replace("CASEID", case_id)
            .replace("SAMPLEID", sample_id)
            .replace("SAMPLENAME", sample_name)
            .replace("PATHTOCASE", case_path)
        )
    return FileDeliverable(**deliverables)
```